# project  prediction of  Iphone buyersUsing Naive bayes and SVM algorithms


In [1]:
#Importing the necessary  Libraries
import  numpy as np
import  pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
#Reading the data 
df=pd.read_csv('iphone_purchase_records.csv')

In [3]:
df.head()

Gender  Age  Salary  Purchase Iphone
0    Male   19   19000                0
1    Male   35   20000                0
2  Female   26   43000                0
3  Female   27   57000                0
4    Male   19   76000                0

In [4]:
# splitting the features and labels  from the data set
features=df.iloc[:,:-1].values
labels=df.iloc[:,-1].values

In [5]:
labels

array([0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1,
       0, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0,
       1, 1, 0, 1, 0, 1, 0, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0,
       1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1,
       0, 1, 0, 1, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 1,

In [6]:
#Encoding the gender  of the  buyers 
from sklearn.preprocessing import LabelEncoder
encoder = LabelEncoder() 
features[:,0] = encoder.fit_transform(features[:,0])



In [8]:
features[:,0]

array([1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0,
       1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1,
       0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1,
       1, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0, 0, 1, 0,
       1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 1, 1, 0,
       0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1,
       1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0,
       1, 0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0,
       0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 0,
       1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1,
       0, 1, 1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
       1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0,
       0, 0, 1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0,

In [12]:
# split the data set into  train and  test  sets, the test is  30% of the original datas  set
from sklearn.model_selection import train_test_split as  tts
features_train,features_test,labels_train,labels_test = tts(features,labels,test_size=0.3,random_state=0)



In [13]:
# scaling the  features of  both train and test datasets
from sklearn.preprocessing import StandardScaler  as scale
scaler  = scale() 
features_train = scaler.fit_transform(features_train) 
features_test  = scaler.transform(features_test)

### classifying  using  Naiive Bayes algorithm 

In [14]:
# training  the  NB   algorithm with the training data 
from sklearn.naive_bayes import GaussianNB
from time import time
clf = GaussianNB()
t0 = time()
clf.fit(features_train,labels_train)
print ("\ntraining time:", round(time()-t0, 3)) # printing the training time 



training time: 0.052


In [15]:
# Making the prediction
t0 = time()
prediction=clf.predict(features_test)
print ("\nprediction time:", round(time()-t0, 3))


prediction time: 0.0


In [16]:
# accuracy of the  algorithm
from sklearn.metrics import accuracy_score
print("Accuracy of the model=",accuracy_score(prediction,labels_test))

Accuracy of the model= 0.9083333333333333


### classifying  using  Support Vector Machine (SVM) algorithm 

In [17]:
# training  the  NB   algorithm with the training data 
from sklearn import svm
clf=svm.SVC()
t0 = time()
clf.fit(features_train,labels_train)
print ("\ntraining  time:", round(time()-t0, 3))


training  time: 0.208


In [18]:
t0 = time()
prediction=clf.predict(features_test)
print ("\nprediction time:", round(time()-t0, 3))


prediction time: 0.001


In [19]:
from sklearn.metrics import accuracy_score
print("Accuracy of the model=",accuracy_score(prediction,labels_test))

Accuracy of the model= 0.925


## Conclusion:
In this  mini Project, I performed  2  classifying algorithms  to predict  the people  who will buy the  IPHone or not based on  some features, namely age, Gender and their salary. The  algorithms used are the Naive bayes  and the  SVM. SVM    showed a better accuracy(92%) compared to the  NB algorithm (90%)
